### PR TITLE
Use cross fetch instead of node-fetch

### PR DESCRIPTION
**Pull request by leeweisberger**
Use cross fetch which works in browser (uses [GitHub's WHATWG Fetch polyfill](https://github.com/github/fetch)) or node (uses node-fetch).
This should just wrap the two libraries and not do anything extra. Later, we might want to consider writing our own impl that just uses browser fetch instead of the polyfill to reduce code size

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +54,7 @@
   },
   "dependencies": {
     "fetch-retry": "5.0.2",
-    "node-fetch": "3.2.3",
+    "isomorphic-fetch": "^3.0.0",
```
Comment (colinking) [inline package.json:57]: pin this for renovate

**Review by colinking — CHANGES_REQUESTED** [review]
Did a quick search. Looks like `cross-fetch` is a better option: https://github.com/lquixada/cross-fetch

`isomorphic-unfetch` is using fairly old versions of `node-fetch`

**Comment by leeweisberger** [comment]
> Did a quick search. Looks like `cross-fetch` is a better option: https://github.com/lquixada/cross-fetch
> 
> `isomorphic-unfetch` is using fairly old versions of `node-fetch`

Fine with this!

**Review by colinking — APPROVED** [review]
LGTM. Any thoughts on how we can include unit tests for the browser env?

**Comment by colinking** [comment]
Also @leeweisberger make sure to test deploying a Node task with this version of the SDK!

**Comment by leeweisberger** [comment]
> LGTM. Any thoughts on how we can include unit tests for the browser env?

In web, we use https://mswjs.io/ which mocks at the network layer and then polyfill in `whatwg-fetch` during tests so that we actually make fetch requests at the server. We could set that up

**Pull request merged**